### ISSUE TITLE
Non-atomic integration tests

### ISSUE THREAD

**Issue by roschaefer** [neo4j-graphql/neo4j-graphql-js#252]
# Problem
@johnymontana explained to me [in this comment](https://github.com/neo4j-graphql/neo4j-graphql-js/pull/247#issuecomment-494579107) that all the tests are expected to pass. What I have to do is to trash the database and re-import the data for each test run.

And indeed: When I run the the integration tests twice, I see all tests pass and next time some of them failing. This is an obvious indicator for so-called [non-atomic tests](https://techbeacon.com/app-dev-testing/how-make-your-test-automation-run-parallel), tests that are not isolated. Running one test case may interfere with another unrelated test case. This is a problem because it deteriorates the quality of the tests, makes flaky behaviour likely and leads to less confidence in the test setup. Plus, it's a not reproducible test setup, it's simply annoying to re-create your database every time.

# Solution

Ideal solution:
* Clean the database after each test run
* For every test case, you have to setup the relevant data e.g. in a `before` block

How could you e.g. test the behaviour of this library, say, on an empty database?

The less optimal solution for now might be to check-in a dump of the database currently in use and load the dump prior to each test case.

**Comment by datenbrei** [comment]
@roschaefer , isn't this very similar to a situation we had a couple of weeks before? don't remember exactly, but there was a setting of the database config to queue the requests serially vs. working multitasked?

**Comment by roschaefer** [comment]
Nope. You talking about jest's `--runInBand` option. [Jest](https://jestjs.io/) is not used as a test runnner in this project, instead it's [ava](https://github.com/avajs/ava).

Plus, the test expect the very same state in the database and it's a lot of data, see below :point_down: It might be a lot of work to refactor the tests to use factories (set up the required data for each test). Then you can do `MATCH(n) DELETE DETACH n;`. If would clean the data after each test run, then the database is empty, which is not what you want. Something like a rollback or a database transaction would be great, but I don't know how to achieve this in our setting.


![Screenshot - 2019-05-22T171806 859](https://user-images.githubusercontent.com/2110676/58186689-ac1cbc80-7cb5-11e9-8496-ae637dd22961.png)

**Comment by johnymontana** [comment]
Ideally, we can restructure these tests to use just a Cypher based fixture for setup, rather than dependency on loading a full database dump

**Comment by danielrearden** [comment]
The full dataset exported as csv is about 149 MB. Would it make sense to just include the csv in the repo as a fixture?

**Comment by michaeldgraham** [comment]
https://github.com/neo4j-graphql/neo4j-graphql-js/issues/608
